design-principles.md

```diff
@@ -0,0 +1,26 @@
+# Tekton Design Principles
+
+## Reusability
+- Existing features should be reused when possible instead of adding new ones.
```

Comment:
Another way to say this (or a complement to this) would be : when implemented a feature, make sure you tried with existing features so that you can make sure you have a clear reason to add this feature.